### ISSUE TITLE
[1.12.2] [Pre_HD_U_D2] Custom Chest GUI Calling issue

### ISSUE THREAD

**Issue by xLastt** [sp614x/optifine#1352]
Hi, it's me again, this is the issue;

When i open a "Custom Chest GUI" from Chest Event called by a Book (Server plugin)

The Custom GUI don't appears
![screenshot_2](https://user-images.githubusercontent.com/26741000/38539028-5adb783a-3c64-11e8-92b1-1857f5bfe4c3.png)
But when i the "Custom Chest GUI" from a normal chest with the custom gui name they work normally...
![screenshot_3](https://user-images.githubusercontent.com/26741000/38539076-95f4753e-3c64-11e8-98d5-38c8b55cb8b2.png)
Okay, the next point is... "The Custom Chest Gui" Works after i open the custom GUI from a chest multiple times and after open the gui called from an event (it's still a Chest GUI) -... The point is after that the Custom GUI opens! But only after that. And after that the rest of sub-menus GUI works
![screenshot_4](https://user-images.githubusercontent.com/26741000/38539202-163f156e-3c65-11e8-8d0a-532b5a68f386.png)
![screenshot_5](https://user-images.githubusercontent.com/26741000/38539214-26bbd62a-3c65-11e8-9ed3-1282c2d52894.png)
![screenshot_6](https://user-images.githubusercontent.com/26741000/38539218-2c586e04-3c65-11e8-9988-674ebfcb7a0a.png)

**Comment by aws404** [comment]
#1111

**Comment by sp614x** [comment]
Duplicate of #1111.

The game does not directly link the chest open action with the GUI opening.
- client sends to server "Click on chest"
- server sends to client "Open GUI XYZ"

To work around this OptiFine caches the last chest click and when the GUI is opened it tries to detect which was the originating action. This may fail when the GUIs are opened in other ways (server plugin).
